domain/resource/service/resource.go

```diff
@@ -163,6 +175,28 @@ func (s *Service) GetApplicationResourceID(
 	return s.st.GetApplicationResourceID(ctx, args)
 }
 
+// GetResourceUUIDByApplicationAndResourceName returns the ID of the application
+// resource specified by natural key of application and resource name.
+//
+// The following error types can be expected to be returned:
+//   - [resourceerrors.ApplicationNotFound] is returned if the application is
+//     not found.
+//   - [resourceerrors.ResourceNotFound] if no resource with name exists for
+//     given application.
```

Comment:
todo: comment does not include the NotValid errors also returned.

Reply:
In my opinion we shouldn't put those in the documentation here, the only place we are expecting to catch those errors is in the tests, the errors here ones that callers of the functions could reasonably expect to detect and act upon.

Reply:
Today that is true, but never say never. If we only list some of the errors it makes for paranoid developers which cannot trust the comments.